src/Commands/Images/GenerateAbstract.php

```diff
@@ -0,0 +1,376 @@
+ * that is bundled with this package in the file LICENSE
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/MIT
+ *
+ * @author Friends of Presta community
+ * @copyright 2019-present Friends of Presta community
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace FOP\Console\Commands\Images;
+
+use Configuration;
+use FOP\Console\Command;
+use Image;
+use ImageManager;
+use ImageType;
+use Language;
+use Module;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+abstract class GenerateAbstract extends Command
+{
+    /** @var array */
+    protected $errors = [];
+
+    /** @var OutputInterface */
+    protected $output;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->output = $output;
+
+        return parent::initialize($input, $output);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('fop:images:generate:' . static::IMAGE_TYPE)
+            ->setDescription('Regenerate ' . static::IMAGE_TYPE . ' thumbnails')
+            ->addArgument(
+                'format',
+                InputArgument::OPTIONAL,
+                'images formats comma separated',
+                'all'
+            )
+            ->addOption(
+                'delete',
+                'd',
+                InputOption::VALUE_OPTIONAL,
+                'Delete current thumbnails',
+                true
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $formats = explode(',', $input->getArgument('format'));
+        $delete = (bool) $input->getOption('delete');
+        $io = new SymfonyStyle($input, $output);
+
+        if (true !== $this->_regenerateThumbnails(static::IMAGE_TYPE, $delete, $formats)) {
+            $io->error('Unable to generate thumbnails');
+
+            return 1;
+        }
+
+        if (count($this->errors)) {
+            $io->error('The generation generate the folowing errors');
+            foreach ($this->errors as $error) {
+                $io->error($error);
+            }
+        }
+
+        $io->success('Thumbnails generated with success for ' . static::IMAGE_TYPE);
+    }
+
+    /**
+     * The whole logic of this file comes from controllers/admin/AdminImagesController.php
+     * with some light adaptations
+     *
+     * @param string $type
+     * @param bool $deleteOldImages
+     * @param array $imagesFormats
+     *
+     * @return bool
+     */
+    protected function _regenerateThumbnails($type = 'all', $deleteOldImages = true, $imagesFormats = ['all'])
```

Comment:
:scream: legacy ! Nice move to mention you copied this code :rofl: :+1: 
I guess we have to deal with it for the moment :/ 

https://github.com/PrestaShop/PrestaShop/blame/d7663195cfc582e1f1285ff03c90eac9859fa9c1/controllers/admin/AdminImagesController.php

Reply:
Yes as this code works no reason to restart from blank for this issue :D

Reply:
the only fact that it is 8 years old is enough to push a refacto of it but i agree with @SebSept statement that it works now and can be improved later (i think the issues coming from it will naturally force a refacto anyway).

And again `_` is a bad practice and discouraged by PSR-2 that we should follow https://www.php-fig.org/psr/psr-2/#43-methods

Reply:
The good way should be to refacto this code in prestashop core first, then, after qa, here

Reply:
This code is duplicated here because it can't be called directly, that's another issue.

Reply:
hum ye ok then what about just extending the class that has these methods and use them where we need them ?